From 5d33c7f1c9ae3812847033d3b14eff4f53c8368b Mon Sep 17 00:00:00 2001
From: Brendan King <Brendan.King@imgtec.com>
Date: Wed, 8 Nov 2017 15:15:20 +0000
Subject: [PATCH 15/71] egl/wayland: post maximum damage when blitting

When blitting, as part of the different display and render GPUs case
when swapping buffers, the blit is done using the full surface
dimensions, ignoring the damage region. This results in corruption of
the non damaged region on screen. Workaround this by posting maximum
damage when blitting.

Upstream-Status: Inappropriate [oe specific]

---
 src/egl/drivers/dri2/platform_wayland.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/egl/drivers/dri2/platform_wayland.c b/src/egl/drivers/dri2/platform_wayland.c
index 909c1a75bc4..aa9634cff2f 100644
--- a/src/egl/drivers/dri2/platform_wayland.c
+++ b/src/egl/drivers/dri2/platform_wayland.c
@@ -1623,7 +1623,8 @@ dri2_wl_swap_buffers_with_damage(_EGLDisplay *disp, _EGLSurface *draw,
    /* If the compositor doesn't support damage_buffer, we deliberately
     * ignore the damage region and post maximum damage, due to
     * https://bugs.freedesktop.org/78190 */
-   if (!n_rects || !try_damage_buffer(dri2_surf, rects, n_rects))
+   if (!dri2_dpy->compat_gpus ||
+       !n_rects || !try_damage_buffer(dri2_surf, rects, n_rects))
       wl_surface_damage(dri2_surf->wl_surface_wrapper, 0, 0, INT32_MAX,
                         INT32_MAX);
 
-- 
2.34.1

